trio/tests/test_signals.py

```diff
@@ -35,23 +34,98 @@
     assert signal.getsignal(signal.SIGILL) is orig
 
 
-def test_catch_signals_wrong_thread():
-    threadqueue = stdlib_queue.Queue()
+async def test_open_signal_receiver():
+    orig = signal.getsignal(signal.SIGILL)
+    with open_signal_receiver(signal.SIGILL) as receiver:
+        # Raise it a few times, to exercise signal coalescing, both at the
+        # call_soon level and at the SignalQueue level
+        signal_raise(signal.SIGILL)
+        signal_raise(signal.SIGILL)
+        await _core.wait_all_tasks_blocked()
+        signal_raise(signal.SIGILL)
+        await _core.wait_all_tasks_blocked()
+        async for signum in receiver:  # pragma: no branch
```

Comment:
Is it worth ensuring that `receiver` only *once* delivers a signal raised multiple times prior to iteration?
I guess I'm just thinking down the road if the implementation is changed (internal `set`-like data structure or whatever) and this behaviour should remain.

Reply:
We are implicitly checking that signals are coalesced, because if they weren't then when we exited the `with` block they'd be redelivered and cause the test process to dump core :-).

It would make sense to test this a little more explicitly... I'm going to add a little private helper to do that.

Reply:
Ahh right of course. Cool, I like explicit :+1: